aot-maven-plugin/src/main/java/com/dylibso/chicory/maven/aot/AotGenMojo.java

```diff
@@ -0,0 +1,275 @@
+package com.dylibso.chicory.maven.aot;
+
+import static com.github.javaparser.StaticJavaParser.parseType;
+
+import com.dylibso.chicory.aot.AotMachine;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.wasm.Parser;
+import com.dylibso.chicory.wasm.types.ValueType;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.expr.ArrayAccessExpr;
+import com.github.javaparser.ast.expr.ArrayCreationExpr;
+import com.github.javaparser.ast.expr.ArrayInitializerExpr;
+import com.github.javaparser.ast.expr.AssignExpr;
+import com.github.javaparser.ast.expr.CastExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.IntegerLiteralExpr;
+import com.github.javaparser.ast.expr.LambdaExpr;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.NullLiteralExpr;
+import com.github.javaparser.ast.expr.SimpleName;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.ast.stmt.ReturnStmt;
+import com.github.javaparser.utils.SourceRoot;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * This plugin generates an invokable library from the compiled Wasm
+ */
+@Mojo(name = "wasm-aot-gen", defaultPhase = LifecyclePhase.GENERATE_RESOURCES, threadSafe = true)
+public class AotGenMojo extends AbstractMojo {
```

Comment:
Not required, but it would be nice to have a test against an example generated class source, so that we can see what the generated code looks like.

Reply:
Do you mean a test "in isolation" other than using `wabt` and `machine-tests`?

Reply:
Yeah, I'm thinking something similar to the annotation processor tests that have the expected generated Java files, or the approval tests for AOT with the expected bytecode. But hopefully with the AOT changes we don't need to generate any code now?

Reply:
Understood, let's keep it for a follow up if you don't mind.
To avoid me forgetting about it, feel free to open an issue, and I'll eventually get there.